File: seed/csharp-model/unions/src/SeedUnions/Types/UnionWithDiscriminant.cs
using System.Text.Json.Serialization;
using SeedUnions;

namespace SeedUnions;

public class UnionWithDiscriminant
{
    public class _Foo
    {
        [JsonPropertyName("_type")]
        public string Type { get; } = "foo";

        [JsonPropertyName("foo")]
        public Foo Foo { get; init; }
    }
    public class _Bar
    {
        [JsonPropertyName("_type")]
        public string Type { get; } = "bar";

        [JsonPropertyName("bar")]
        public Bar Bar { get; init; }
    }
}

File: seed/swift-sdk/unions/Sources/Schemas/UnionWithDiscriminant.swift
public enum UnionWithDiscriminant: Codable, Hashable, Sendable {
    case foo(Foo)
    case bar(Bar)

    public struct Foo: Codable, Hashable, Sendable {
        public let foo: Foo

        public init(foo: Foo) {
        }

        private enum CodingKeys: String, CodingKey {
        }
    }

    public struct Bar: Codable, Hashable, Sendable {
        public let bar: Bar

        public init(bar: Bar) {
        }

        private enum CodingKeys: String, CodingKey {
        }
    }
}